repository/gorm/message.go

```diff
@@ -289,49 +291,115 @@ func (repo *Repository) GetDeletedMessagesAfter(after time.Time, limit int) (mes
 	return
 }
 
-// SetMessageUnread implements MessageRepository interface.
-func (repo *Repository) SetMessageUnread(userID, messageID uuid.UUID, noticeable bool) error {
-	if userID == uuid.Nil || messageID == uuid.Nil {
+// SetMessageUnreads implements MessageRepository interface.
+func (repo *Repository) SetMessageUnreads(userNoticeableMap map[uuid.UUID]bool, messageID uuid.UUID) error {
+	if messageID == uuid.Nil {
 		return repository.ErrNilID
 	}
+	if len(userNoticeableMap) == 0 {
+		return nil
+	}
+
+	userIDs := make([]uuid.UUID, 0, len(userNoticeableMap))
+	for userID := range userNoticeableMap {
+		if userID == uuid.Nil {
+			return repository.ErrNilID
+		}
+		userIDs = append(userIDs, userID)
+	}
+
+	// 流れ
+	// unreadテーブルにuserID, messageIDが存在するか確認する
+	// 存在しないユーザーについて
+	//   - messageIDのメッセージが存在するか確認
+	//   - 存在する場合、unreadテーブルにuserID, messageID, channelID, noticeable, messageCreatedAtを登録
+	//   - 存在しない場合、エラーを返す
+	// 存在するユーザーについて
+	//   - noticeableを更新
+	//   - created フラグを立てない
+	// DB 更新後、 unread テーブルにレコードが存在しなかったユーザーだけイベントを発行する
```

Comment:
createdフラグのコメントの消し忘れ
あと`messageIDのメッセージが存在するか確認`はしてなさそう、、？

```suggestion
	// 流れ
	// unreadテーブルにuserID, messageIDが存在するか確認する
	// 存在しないユーザーについて
	//   - unreadテーブルにuserID, messageID, channelID, noticeable, messageCreatedAtを登録
	//   - 存在しない場合、エラーを返す
	// 存在するユーザーについて
	//   - noticeableを更新
	// DB 更新後、 unread テーブルにレコードが存在しなかったユーザーだけイベントを発行する
```